src/emqx_mod_topic_metrics.erl

```diff
@@ -182,6 +187,25 @@ val(Topic, Metric) ->
 rate(Topic, Metric) ->
     gen_server:call(?MODULE, {get_rate, Topic, Metric}).
 
+last_short(LastSpeed, CurSpeed) ->
+    mma(?SPEED_AVERAGE_WINDOW_SIZE, LastSpeed, CurSpeed).
+
+last_medium(LastSpeed, CurSpeed) ->
+    mma(?SPEED_MEDIUM_WINDOW_SIZE, LastSpeed, CurSpeed).
+
+last_long(LastSpeed, CurSpeed) ->
+    mma(?SPEED_LONG_WINDOW_SIZE, LastSpeed, CurSpeed).
+
+last_stats_all(LastSpeed, CurSpeed) ->
+        #{
+            short => last_short(LastSpeed, CurSpeed),
+            medium => last_medium(LastSpeed, CurSpeed),
+            long => last_long(LastSpeed, CurSpeed)
+         }.
+
+rates(LastSpeed, CurSpeed) ->
```

Comment:
the arguments should be the same as `rate/2`.
arg1 is the topic, and arg2 is the metric name.

i.e. the values are stored inside gen_server state,
if a caller would already knew the values, there is no need to call this gen_server to do the calculation.